lib/ProofSet.js

```diff
@@ -306,9 +306,10 @@ async function _verify({
   document, suites, proofSet, purpose,
   documentLoader, expansionMap, compactProof}) {
   // filter out matching proofs
-  const matches = proofSet.filter(proof => purpose ?
+  const result = await Promise.all(proofSet.map(proof => purpose ?
```

Comment:
`purpose` is always defined so we can remove the conditional and just always call `purpose.match`. I think this was some noise left over from legacy code.

Reply:
Done.